xoinvader/tests/test_background.py

```diff
@@ -0,0 +1,128 @@
+
+    with pytest.raises(ValueError):
+        load_chunks("xoinvader/tests/chunk_duplicate_names.bg")
+
+    with pytest.raises(ValueError):
+        load_chunks("xoinvader/tests/no_chunks.bg")
+
+    c = load_chunks("xoinvader/tests/chunk_normal.bg")
+    assert(len(c) == 3)
+    assert(c[0][0] == "qweqwe")
+    assert(c[0].length == 1)
+
+    d = load_chunks("xoinvader/tests/chunk_normal.bg", 3)
+    assert(d[0][0] == "qwe")
+
+def test_background():
+    Settings.layout.field.edge.x = 3
+    Settings.layout.field.edge.y = 2
+
+    b = Background()
+    assert not b.loop
+    assert not b.loop_all
+    assert not b.speed
+    assert not b.chunks
+    assert not b.background
+    assert not b._background_surface
+    assert not b._current_chunk
+    assert not b._current_chunk_num
+    assert not b._chunk_line
+    assert not b._ticks_since_last_update
+
+    assert b._w == Settings.layout.field.edge.x
+    assert b._h == Settings.layout.field.edge.y
+
+    assert b._advance_chunk(1) == "   "
+
+    b = Background("xoinvader/tests/chunk_normal.bg")
+    assert len(b.chunks) == 3
+    assert len(b.chunks[2].lines) == 3
+
+    b.start()
+    assert b._current_chunk is b.chunks[0]
+    assert b.background == ["   ",
+                            "   "]
+
+    b.start(True)
+    assert b._chunk_line == 1
+    assert b._current_chunk is b.chunks[1]
+    assert b.background == ["qwe",
+                            "asd"]
+    assert b._chunk_line == 1
+    assert b._advance_chunk(1) == "zxc"
+    assert b._chunk_line == 1
+    assert b._advance_chunk(-1) == "123"
+    assert b._chunk_line == 0
+    assert b._advance_chunk(1) == "zxc"
+    assert b._chunk_line == 1
+    b.loop = True
+    assert b._advance_chunk(1) == "123"
+    assert b._advance_chunk(1) == "!@#"
+    assert b._advance_chunk(-1) == "zxc"
+    assert b._advance_chunk(-1) == "!@#"
+    b.loop = False
+    b.loop_all = True
+    assert b._advance_chunk(1) == "123"
+    assert b._advance_chunk(1) == "!@#"
+    assert b._advance_chunk(1) == "qwe"
+    assert b._advance_chunk(1) == "asd"
+    assert b._advance_chunk(1) == "zxc"
+    assert b._advance_chunk(-1) == "123"
+    assert b._advance_chunk(-1) == "zxc"
+    assert b._advance_chunk(-1) == "asd"
+    assert b._advance_chunk(-1) == "qwe"
+    assert b._advance_chunk(-1) == "!@#"
+    b.loop_all = False
+    assert b._advance_chunk(1) == "123"
+    assert b._advance_chunk(1) == "!@#"
+    assert b._advance_chunk(1) == "   "
+
+    P, S = b.get_render_data()
+    assert P == [Point(0,0)]
+    assert next(S) == (Point(0,0,0), "q", None)
+    assert next(S) == (Point(1,0,0), "w", None)
+    assert next(S) == (Point(2,0,0), "e", None)
+    assert next(S) == (Point(0,1,0), "a", None)
+    assert next(S) == (Point(1,1,0), "s", None)
+    assert next(S) == (Point(2,1,0), "d", None)
+    with pytest.raises(StopIteration):
+        next(S)
+
+    assert not b.update()
+    b.speed = 40
+    bg = copy(b.background)
```

Comment:
oh, that's tricky.
if object holds another mutables, copy will create links to them.
For completely copy of object you can use deepcopy from copy module.